### PR TITLE
Set localStorage settings displayName with shibboleth attribute

### PR DESCRIPTION
**Pull request by daimoc**
Set the shibboleth displayName or email attribute to the localSorage variable features/base/settings.displayName to use it in JItsi-Meet UI.

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/auth/ShibbolethHandler.java [hunk 1]
```diff
@@ -242,8 +242,23 @@ private void doHandle(
...
-
+                " console.info('sessionID :' + sessionId);\n" +
+                " var displayName = '" + displayName + "';\n" +
+                " console.info('displayName :' + displayName);\n" +
+                " var settings = localStorage.getItem('features/base/settings');\n" +
+                " console.info('settings :' + settings);\n" +
+                " if (settings){\n" +
+                "     try {\n" +
+                "	        var settingsObj = JSON.parse(settings);\n" +
+                "	        if ( settingsObj != null && settingsObj.displayName != null && settingsObj.displayName == \"\" ) {\n" +
```
Comment (paweldomas) [inline src/main/java/org/jitsi/jicofo/auth/ShibbolethHandler.java]: do you think a truthy check for `settingsObj.displayName` would be good enough ?
`if (settingsObj != null && !settingsObj.displayName)`

Comment (daimoc) [reply src/main/java/org/jitsi/jicofo/auth/ShibbolethHandler.java]: You are perfectly right Pawel, a truthy check is enough. 
Thanks.

**Comment by daimoc** [comment]
Hello, any news about this PR approval ?

**Comment by bertvandepoel** [comment]
@SimonMaenaut and I are currently working on improving the user experience of those who login through Shibboleth. We are introducing some changes to the displayName in that PR and would also like to include your change in it, but of course with a change on our end as the displayName variable has changed in our code to be more generic (we wish to support a wider range of Shibboleth attributes). Have you signed the CLA and are you OK with us using a portion of your code? Or can we include your commit in our PR somehow and then apply our change over it (using cherrypicking)?

**Comment by daimoc** [comment]
Hi @bertvandepoel, I've already sign the CLA  and of course I'm OK if you want to use the code I made for this PR. I made this PR a long time ago and I don't know why it didn't merge.
Any improvements in Shibboleth code are welcome, and thanks for your work on recent fix.
Sorry for my late response but I think everyone working with Jitsi has a lot of work now.

**Comment by BogdanRudas** [comment]
How can I evaluate this PR ? Are there any build instructions or custom build?
I'm using similar setup with OIDC apache2 module to enforce authentication for meeting starters and passing email and displayName as HTTP headers to Jicofo.

**Comment by paweldomas** [comment]
@daimoc I'm not testing this PR as we're not using this feature anywhere, so it's up to the community to keep it working. I will merge it once someone confirms that it's working

**Comment by creativeguitar** [comment]
Lots of people, including myself, are interested in this update. When is it merging?

**Comment by BogdanRudas** [comment]
Works for me (I'm not using Shibboleth, but passing required info as HTTP headers to jicofo).

**Pull request merged**